### DISCUSSION TITLE
excessive download of data

### DISCUSSION THREAD

**Discussion by koshikas** [spacecowboy/Feeder#339] [General]
i still cant say how it happens, but sometimes all of a sudden, feeder consumes in excess of 1GB(sometimes even 2GB) of data for no reason. i havent subscribed to any heavy feeds as such. 

![image](https://github.com/user-attachments/assets/f19e858d-86a9-4ffa-a8a8-b1d0e687ea22)

**Comment by spacecowboy** [comment]
That is quite crazy.

A feed is only a couple of 100 kilobytes, max, typically. So it's not that.

Have you enabled "full article by default" on any feeds? I would guess that is where it happens.

**Reply by koshikas** [reply]
i agree. almost always a sync to all feeds would require a data budget as in the following screenshot. its just that i found out (2 times) when there were massive downloads from "feeder" for no reason. 

double checked settings, i have not set "full article by default" in any of my feeds. i will hopefully take logs next time to see if i can give more context if this ever happens again. i did have sync all feeds automatically on app start when this happened, which i've since disabled, so now the feeds are synced manually. let me fiddle around and see what i can find. thanks

![SAVE_20240802_121203](https://github.com/user-attachments/assets/82a13c42-3321-4602-92b6-a65d7f37dbdf)

**Reply by spacecowboy** [reply]
It's not the first time I've heard of this. But have no idea what combination of settings etc that can trigger it.

But you had sync on startup enabled, but NOT automatic sync every N hours?

**Reply by koshikas** [reply]
yes, sync on startup enabled, but NOT automatic sync every N hours

**Reply by koshikas** [reply]
so it strikes again, this time i had sync setup to be done manually, i did not rush to sync, but somehow it happened again.
![IMG_20240820_182256-01](https://github.com/user-attachments/assets/f7f13793-7d8c-4cda-b590-5380a25c02ed)

**Reply by spacecowboy** [reply]
Are you using the `Device Sync` feature?

**Reply by koshikas** [reply]
nope

**Reply by spacecowboy** [reply]
Crazy idea. You can't by any chance see traffic statistics on your WiFi router by any chance? See if there is a particular website related to the traffic?

**Reply by koshikas** [reply]
let me do that going forward, will report back with my findings

**Reply by koshikas** [reply]
so i did manage to get some hold of the situation. i ran a "pcap" of Feeder, and found sourceforge's RSS feeds to be the culprit in my case. seems like the RSS feed somehow triggers a data downstream for no reason from one of the mirrors where its file are hosted. i've since unsubscribed from all sourceforge feeds and it seems the massive data downloads are no more.

i've attached a PCAP report last time i encountered the issue with sourceforges's feeds, for your reference. from my limited investigations i think it was this feed (in OPML form) that caused it;
outline xmlUrl="https://sourceforge.net/projects/crystaldiskinfo/rss?path=/" title="CrystalDiskInfo"

[PCAPdroid_31_Aug_23_59_43.csv](https://github.com/user-attachments/files/17203522/PCAPdroid_31_Aug_23_59_43.csv)

**Comment by marzagheddon** [comment]


**Reply by spacecowboy** [reply]
Please be constructive in your remarks.

**Comment by Mr-Bajs** [comment]
Have some high Wi-Fi data usage too. Atleast it seems that to me.

- Jun 9-Jul 7 - Foreground 0,165 GB Background 0,99 GB
- Jul 7-Aug 4 - Foreground 1,11 GB Background 4,59 GB
- Aug 4-Sept 1 - Foreground 0,89 GB Background 3,72 GB
- Sep 1-Sep 29 - Foreground 0,98 GB Background 4,42 GB

**Comment by spacecowboy** [comment]
Again, I have no idea why this is happening. It doesn't happen to myself. So I really need help to pin this problem down.

I have enabled issues on the project now. Would appreciate if someone could make a bug report and fill in the details there

**Reply by Mr-Bajs** [reply]
Just a wild guess. 
Could the data somehow go trough Feeder when you open an article with web view and then any continous browsing in browser somehow be considered be with Feeder anyhow?

**Reply by spacecowboy** [reply]
Sounds unlikely, but have no idea if it's possible to be honest.
